mindsdb/integrations/handlers/derby_handler/derby_handler.py

```diff
@@ -0,0 +1,257 @@
+            log.logger.error(f"Error while connecting to {self.database}, {e}")
+
+        return self.connection
+
+
+    def disconnect(self):
+        """ Close any existing connections
+        Should switch self.is_connected.
+        """
+        if self.is_connected is False:
+            return
+        try:
+            self.connection.close()
+            self.is_connected=False
+        except Exception as e:
+            log.logger.error(f"Error while disconnecting to {self.database}, {e}")
+
+        return 
+
+
+    def check_connection(self) -> StatusResponse:
+        """ Check connection to the handler
+        Returns:
+            HandlerStatusResponse
+        """
+        responseCode = StatusResponse(False)
+        need_to_close = self.is_connected is False
+
+        try:
+            self.connect()
+            responseCode.success = True
+        except Exception as e:
+            log.logger.error(f'Error connecting to database {self.database}, {e}!')
+            responseCode.error_message = str(e)
+        finally:
+            if responseCode.success is True and need_to_close:
+                self.disconnect()
+            if responseCode.success is False and self.is_connected is True:
+                self.is_connected = False
+
+        return responseCode
+
+
+    def native_query(self, query: str) -> StatusResponse:
+        """Receive raw query and act upon it somehow.
+        Args:
+            query (Any): query in native format (str for sql databases,
+                dict for mongo, etc)
+        Returns:
+            HandlerResponse
+        """
+        need_to_close = self.is_connected is False
+        conn = self.connect()
+        with conn.cursor() as cur:
+            try:
+                cur.execute(query)
+                if cur.description:
+                    result = cur.fetchall() 
+                    response = Response(
+                        RESPONSE_TYPE.TABLE,
+                        data_frame=pd.DataFrame(
+                            result,
+                            columns=[x[0] for x in cur.description]
+                        )
+                    )
+                else:
+                    response = Response(RESPONSE_TYPE.OK)
+                self.connection.commit()
+            except Exception as e:
+                log.logger.error(f'Error running query: {query} on {self.database}!')
+                response = Response(
+                    RESPONSE_TYPE.ERROR,
+                    error_message=str(e)
+                )
+                self.connection.rollback()
+
+        if need_to_close is True:
+            self.disconnect()
+
+        return response
+
+    
+    def query(self, query: ASTNode) -> StatusResponse:
+        """Render and execute a SQL query.
+
+        Args:
+            query (ASTNode): The SQL query.
+
+        Returns:
+            Response: The query result.
+        """
+        if isinstance(query, ASTNode):
```

Comment:
Did you check if there is a sqlalchemy dialect for derby available?

Reply:
I did check this, but there is no sqlalchemy dialect for derby right now. Refer: [Link](https://github.com/sqlalchemy/sqlalchemy/issues/4839)